Source/utils/log.hpp

```diff
@@ -25,25 +25,40 @@ enum class LogCategory {
 constexpr auto defaultCategory = LogCategory::Application;
 
 enum class LogPriority {
-    Verbose = SDL_LOG_PRIORITY_VERBOSE,
-    Debug = SDL_LOG_PRIORITY_DEBUG,
-    Info = SDL_LOG_PRIORITY_INFO,
-    Warn = SDL_LOG_PRIORITY_WARN,
-    Error = SDL_LOG_PRIORITY_ERROR,
-    Critical = SDL_LOG_PRIORITY_CRITICAL,
+	Verbose = SDL_LOG_PRIORITY_VERBOSE,
+	Debug = SDL_LOG_PRIORITY_DEBUG,
+	Info = SDL_LOG_PRIORITY_INFO,
+	Warn = SDL_LOG_PRIORITY_WARN,
+	Error = SDL_LOG_PRIORITY_ERROR,
+	Critical = SDL_LOG_PRIORITY_CRITICAL,
 };
 
+namespace detail {
+
+template <typename... Args>
+std::string format(const char *fmt, Args &&... args)
+{
+	try {
+		return fmt::format(fmt, std::forward<Args>(args)...);
+	} catch (const fmt::format_error &e) {

```

Comment:
Won't this cause us to mask real issues that need to be fixed ASAP? Is there a real situation where it would make sense to ignore these errors? Aren't they all actual issues?

Not a big fan of swallowing exceptions in this case if the above is true.

Reply:
I would probably go with `app_fatal()` instead of logging. I'm pretty sure that the case we are doing this for should never happen, so would be good to just get the tests to report it back, and most won't be looking in the log.

Reply:
> Wait, fmt makes use of exceptions :(

Well exceptions are the [official](https://isocpp.org/wiki/faq/exceptions#why-exceptions) way of signaling errors in C++, so they are quite a common sight, even if you don't like them :stuck_out_tongue: 

> Won't this cause us to mask real issues that need to be fixed ASAP? Is there a real situation where it would make sense to ignore these errors? Aren't they all actual issues?
> Not a big fan of swallowing exceptions in this case if the above is true.

> I would probably go with app_fatal() instead of logging. I'm pretty sure that the case we are doing this for should never happen, so would be good to just get the tests to report it back, and most won't be looking in the log.

I agree that this is an important issue. I propose to log it anyway but also to call `app_fatal()` so if someone sends a log file to us it will be easier to diagnose the issue. Generally logging more is good.

> On a side note could you check that we can build with exceptions disabled. Requiring exceptions is preventing me from building on my powerpc using newer compilers:
fmtlib/fmt@8b76e97

I will use the ugly FMT macros then.

I didn't know there were systems that didn't support exceptions. That is very odd. Do you know why?

Reply:
I think maybe the easy route is to make `UiOkDialog()` always log, not just when there is no display.